### PR TITLE
GAWB-879: use a service account for orch-to-rawls admin communication

### PR DESCRIPTION
**Pull request by davidangb**
See also:
- https://github.com/broadinstitute/firecloud-develop/pull/114
- https://github.com/broadinstitute/firecloud-develop/pull/115

Beyond those config changes, we need to manually add the service account to LDAP and to the admins group in each environment. I have done this for dev, but no others. Once this hits dev I'll either do it for other environments, or pester someone who has more access than me to do it.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: Tell tech lead that the PR exists if s/he wants to look at it
- [ ] **Submitter**: Anoint a lead reviewer (LR). **Assign PR to LR**
- [ ] **LR**: Initial review by LR and others.
- [ ] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] **Tech lead**:  sign off
- [x] **LR**: sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala [hunk 1]
```diff
@@ -23,11 +23,12 @@ object HttpGoogleServicesDAO {
...
-  val gcsFullControl = "https://www.googleapis.com/auth/devstorage.full_control"
-  val computeFullControl = "https://www.googleapis.com/auth/compute"
-  // TODO: remove compute?
-  val scopes = Seq(gcsFullControl,computeFullControl,"profile","email")
+  // the minimal scopes needed to get through the auth proxy and populate our UserInfo model objects
+  val authScopes = Seq("profile","email")
+  // the minimal scope to read from GCS
+  val storageReadOnly = Seq(StorageScopes.DEVSTORAGE_READ_ONLY)
+  // the scopes we request for the end user during interactive login. TODO: remove compute?
+  val userLoginScopes = Seq(StorageScopes.DEVSTORAGE_FULL_CONTROL,"https://www.googleapis.com/auth/compute") ++ authScopes
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala:31]: My reading here is that user login scopes have not changed.  Can you confirm this?

File: src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala [hunk 2]
```diff
@@ -111,10 +107,13 @@ object HttpGoogleServicesDAO {
...
 
   def getAdminUserAccessToken = {
-    val googleCredential = new GoogleCredential.Builder().setTransport(httpTransport)
+    val googleCredential = new GoogleCredential.Builder()
+      .setTransport(httpTransport)
       .setJsonFactory(jsonFactory)
-      .setClientSecrets(refreshTokenClientSecrets)
-      .build().setFromTokenResponse(new TokenResponse().setRefreshToken(adminUserRefreshToken))
+      .setServiceAccountId(pemFileClientId)
+      .setServiceAccountScopes(authScopes) // use the smallest scope possible
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala:114]: good

**Comment by jmthibault79** [comment]
👍

File: src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala (hunk 1 above)
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala:31]: Correct - nothing has changed, I just tried to clean up/reorganize the code a bit.

**Pull request merged**